### PR TITLE
[SPARK-38746][SQL][TESTS] Move the tests for `PARSE_EMPTY_STATEMENT` to QueryParsingErrorsSuite

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims to move tests for the error class `PARSE_EMPTY_STATEMENT` from ErrorParserSuite to QueryParsingErrorsSuite., it's a followup of SPARK-37935.

### Why are the changes needed?
To improve code maintenance.
 
### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
By running the moved tests:
```
$ build/sbt "sql/testOnly *QueryParsingErrorsSuite*"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala [hunk 1]
```diff
@@ -405,4 +405,47 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
           |-----------------------------^^^
           |""".stripMargin)
   }
+
+  test("PARSE_EMPTY_STATEMENT: ") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:409]: nit:
```suggestion
  test("PARSE_EMPTY_STATEMENT: empty input") {
```

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala [hunk 2]
```diff
@@ -405,4 +405,47 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
...
+      sqlText = "   ",
+      errorClass = "PARSE_EMPTY_STATEMENT",
+      sqlState = "42000",
+      message =
+        """
+          |Syntax error, unexpected empty statement(line 1, pos 3)
+          |
+          |== SQL ==
+          |   """.stripMargin +
+          """
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:433]: Can't you from a single string?

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala [hunk 3]
```diff
@@ -405,4 +405,47 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
...
+      sqlText = " \n",
+      errorClass = "PARSE_EMPTY_STATEMENT",
+      sqlState = "42000",
+      message =
+        """
+          |Syntax error, unexpected empty statement(line 2, pos 0)
+          |
+          |== SQL ==
+          | """.stripMargin +
+          """
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:447]: The same as above - form one string, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:409]: ok

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala (hunk 2 above)
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala]: ok

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
Waiting for https://github.com/panbingkun/spark/actions/runs/2182508354

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**